testfixtures/logcapture.py

```diff
@@ -73,11 +86,53 @@ def atexit(cls):
                 '%s' % ('\n'.join((str(i.names) for i in cls.instances)))
                 )
 
+    def __len__(self):
+        return len(self.records)
+
+    def __getitem__(self, key):
+        return self._rec_2_obj(self.records[key])
+
+    def __contains__(self, what):
+        for i, item in enumerate(self):
+            if what == item:
+                self.records[i].is_asserted = True
+                return True
+        return False
```

Comment:
nitpick: the `return False` isn't required here, default return is `None` which has the same effect.